### PR TITLE
Fix logging issues in the debugger.

### PR DESCRIPTION
**Pull request by zherczeg**
- Properly handle logging during transport close
 - Properly display data during parse

Furthermore hide global functions of the debugger.

### CODE REVIEW

**Comment by zherczeg** [comment]
Good point. The internal function hiding is done in #2484.

**Comment by akosthekiss** [comment]
Isn't this now pending on #2484? The current version of this PR still contains rewrites that are happening in the referenced PR.

**Comment by zherczeg** [comment]
This should be landed after that PR.

**Comment by LaszloLango** [comment]
#2484 is landed, please rebase the PR.

**Comment by zherczeg** [comment]
Patch updated. Now debugger print works correctly with \0 characters.

File: jerry-ext/handler/handler-print.c [hunk 1]
```diff
@@ -56,24 +56,24 @@ jerryx_handler_print (const jerry_value_t func_obj_val, /**< function object */
...
-      if (arg_index != 0)
-      {
-        jerryx_port_handler_print_char (' ');
-      }
-
+      const jerry_size_t substr_max_size = 256;
       jerry_size_t substr_size;
       jerry_length_t substr_pos = 0;
-      jerry_char_t substr_buf[256];
+      jerry_char_t substr_buf[substr_max_size];
```
Comment (LaszloLango) [inline jerry-ext/handler/handler-print.c]: Plase use `JERRY_VLA` to fix the windows build.

Comment (zherczeg) [reply jerry-ext/handler/handler-print.c]: But this is a fixed size array :(

Comment (LaszloLango) [reply jerry-ext/handler/handler-print.c]: What about using #define instead of a const variable?

**Pull request merged**